tutorial/live_updates.py

```
@@ -170,4 +170,110 @@ def serve_layout():
 You can combine this with [time-expiring caching](/performance) and serve
 a unique `layout` every hour or every day and serve the computed `layout`
 from memory in between.
-''')]
+'''),
+
+          dcc.Markdown('''
+
+***
+
+## Dynamically Create Layout for Multi-page App Validation
+
+Dash applies validation to your callbacks, which performs checks such as 
+validating the types of callback arguments and checking to see whether the 
+specified Input and Output components actually have the specified properties. 
+
+For full validation, all components within your callback must therefore appear 
+in the initial layout of your app, and you will see an error if they do not.
+However, in the case of more complex Dash apps that involve dynamic modification 
+of the layout (such as multi-page apps), not every component appearing in your 
+callbacks will be included in the initial layout.
+
+Since this validation is done before flask has any request context, you can create 
+a layout function that checks `flask.has_request_context()` and returns a complete
+layout to the validator if there is no request context and returns the incomplete
+index layout otherwise:
+'''),
+    dcc.SyntaxHighlighter(
+        '''
+import dash
+import dash_html_components as html
+import dash_core_components as dcc
+from dash.dependencies import Input, Output, State
+
+import flask
+
+app = dash.Dash(__name__, external_stylesheets=['https://codepen.io/chriddyp/pen/bWLwgP.css'])
+
+layout_index = html.Div([
+    dcc.Tabs(id="tabs", value='tab-1', children=[
```

I think an example with `dcc.Location` multi-page app would be better here, from what I've seen that pattern almost always suppress the exceptions. It is set in our multi-page examples https://dash.plot.ly/urls.